File: core-arch/src/main/java/com/huyingbao/core/arch/utils/RxAndroidInjection.java
package com.huyingbao.core.arch.utils;

import android.app.Application;

import androidx.annotation.NonNull;

import dagger.android.AndroidInjector;
import dagger.android.HasAndroidInjector;

import static dagger.internal.Preconditions.checkNotNull;

/**
 * 公用Dagger.Android依赖注入方法
 */
public final class RxAndroidInjection {
    public static void inject(@NonNull Object object, @NonNull Application application) {
        AndroidInjector<Object> injector;
        if (application instanceof HasAndroidInjector) {
            injector = ((HasAndroidInjector) application).androidInjector();
            checkNotNull(injector, "%s.androidInjector() returned null", application.getClass());
        } else {
            throw new RuntimeException(
                    String.format(
                            "%s does not implement %s",
                            application.getClass().getCanonicalName(),
                            HasAndroidInjector.class.getCanonicalName()));
        }
        injector.inject(object);
    }

    private RxAndroidInjection() {
    }
}

File: core-arch/src/main/java/com/huyingbao/core/arch/utils/RxAndroidInjection.kt
package com.huyingbao.core.arch.utils

import dagger.android.HasAndroidInjector
import dagger.internal.Preconditions.checkNotNull

/**
 * Dagger.Android 中[dagger.android.AndroidInjection.inject]扩展为public
 *
 * Created by liujunfeng on 2019/1/1.
 */
object RxAndroidInjection {
    /**
     * Dagger.Android 中[dagger.android.AndroidInjection.inject]扩展为public
     *
     * @param target      需要被依赖注入的类
     */
    fun inject(target: Any, hasAndroidInjector: HasAndroidInjector) {
        val androidInjector = hasAndroidInjector.androidInjector()
        checkNotNull(
                androidInjector,
                "%s.androidInjector() returned null",
                hasAndroidInjector.javaClass)
        androidInjector.inject(target)
    }
}
